lisa/transformers/kernel_installer.py

```diff
@@ -140,37 +142,17 @@ def _internal_run(self) -> Dict[str, Any]:
             if installer.runbook.source != "linux-image-azure-fde":
                 posix = cast(Posix, node.os)
                 posix.replace_boot_kernel(installed_kernel_version)
-            else:
-                efi_files = node.execute(
-                    "ls -t /usr/lib/linux/efi/kernel.efi-*-azure-cvm",
-                    sudo=True,
-                    shell=True,
-                    expected_exit_code=0,
-                    expected_exit_code_failure_message=(
-                        "fail to find kernel.efi file for kernel type "
-                        " linux-image-azure-fde"
-                    ),
-                )
-                efi_file = efi_files.stdout.splitlines()[0]
-                node.execute(
-                    (
-                        "cp /boot/efi/EFI/ubuntu/grubx64.efi "
-                        "/boot/efi/EFI/ubuntu/grubx64.efi.bak"
-                    ),
-                    sudo=True,
-                )
-                node.execute(
-                    f"cp {efi_file} /boot/efi/EFI/ubuntu/grubx64.efi",
-                    sudo=True,
-                    shell=True,
-                )
 
         self._log.info("rebooting")
         node.reboot()
+        kernel_version_after_install = uname.get_linux_information(force_run=True)
         self._log.info(
             f"kernel version after install: "
-            f"{uname.get_linux_information(force_run=True)}"
+            f"{kernel_version_after_install}",
         )
+        assert_that(
+            kernel_version_after_install, "Kernel installation Failed"
+        ).is_equal_to(kernel_version_before_install)
```

Comment:
@squirrelsc long time ago, we had a discussion on this check, do you think we can add the check?

Reply:
Yes, it looks good to me.